gateway-developer-plugin/src/main/java/com/ca/apim/gateway/cagatewayconfig/BuildEnvironmentBundleTask.java

```diff
@@ -52,29 +51,41 @@ DirectoryProperty getInto() {
         return into;
     }
 
+    @Input
+    @Optional
+    Property<Map> getEnvironmentConfig() {
+        return environmentConfig;
+    }
+
     @InputDirectory
+    @Optional
     DirectoryProperty getConfigFolder() {
         return configFolder;
     }
 
     @Input
+    @Optional
     Property<String> getConfigName() {
         return configName;
     }
 
     @TaskAction
     public void perform() {
         final EnvironmentBundleCreator environmentBundleCreator = getInstance(EnvironmentBundleCreator.class);
-        final List<File> metaDataFiles = collectFiles(into.getAsFile().get().getPath(), YML_EXTENSION);
-        if(metaDataFiles.isEmpty()) {
+        final List<File> metaDataFiles = collectFiles(into.getAsFile().get().getPath(), JsonFileUtils.METADATA_FILE_NAME_SUFFIX);
+        if (metaDataFiles.isEmpty()) {
             throw new MissingEnvironmentException("Metadata file does not exist.");
         }
-        metaDataFiles.stream().forEach(metaDataFile-> {
-            final Pair<String, Map<String, String>> bundleEnvironmentValues = environmentConfigurationUtils.parseBundleMetadata(metaDataFile, configFolder.getAsFile().get());
+
+        metaDataFiles.stream().forEach(metaDataFile -> {
+            final Pair<String, Map<String, String>> bundleEnvironmentValues = environmentConfigurationUtils.parseBundleMetadata(metaDataFile, configFolder.getAsFile().getOrNull());
             if (null != bundleEnvironmentValues) {
-                final String bundleFileName = bundleEnvironmentValues.getLeft() + "." + configName.get() + ".environment.bundle";
+                final String bundleFileName = bundleEnvironmentValues.getLeft() + "-" + configName.get() + "env.install.bundle";
```

Comment:
Can you revert this change please. It will be a conflict with #234

Reply:
@nikhilahuja , do not worry about this, I will merge your PR once it is approved